src/core/common/drivers/base.py

```diff
@@ -153,6 +153,12 @@ def close(self):
             self.connection.close()
         self.connection = None
 
+    def test(self):
```

Comment:
This doesn't actually test the connection. It tests whether we have previously tried to run a query successfully. 

We should be running an actual query here as a test and verifying the response. E.g. SELECT 1;